githublib.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+#
+# This file is part of the github2jira project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2021 Red Hat, Inc.
+#
+
+import time
+import requests
+
+from datetime import datetime
+
+from config import Config
+
+SECONDS_PER_WEEK = 604800
+# max github pages to process
+GITHUB_MAX_PAGES = 20
+# process upto x weeks back
+MAX_DELTA_WEEKS = 4
+
+
+class GithubConfig(Config):
+    def __init__(self):
+        super().__init__(
+            ["GITHUB_TOKEN", "GITHUB_OWNER", "GITHUB_REPO", "GITHUB_LABEL"]
+        )
+
+
+class Issue:
+    def __init__(self, issue):
+        self.issue = issue
+
+    def filter(self, expected_label):
+        if expected_label == "":
+            return True
+        for label in self.issue["labels"]:
+            if label["name"] == expected_label:
+                return True
+        return False
+
+    def age_relevant(self, max_delta):
+        epoch_time_now = int(time.time())
+        TIME_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+        timestamp = self.issue["created_at"]
+        epoch = int(datetime.strptime(timestamp, TIME_FORMAT).timestamp())
+        return (epoch_time_now - epoch) < (max_delta * SECONDS_PER_WEEK)
+
+    @property
+    def repo(self):
+        return self.issue["html_url"].split("/")[4]
+
+    @property
+    def id(self):
+        return self.issue["number"]
+
+    @property
+    def url(self):
+        return self.issue["html_url"]
+
+    @property
+    def title(self):
+        return self.issue["title"]
+
+
+class Github:
+    def __init__(self, cfg):
+        self.owner = cfg.vars["GITHUB_OWNER"]
```

Comment:
It looks like `"GITHUB_OWNER"` (and the others) needs to be known in more than one place. We need to aim to know its name only once.